op-node/p2p/peer_gater.go

```diff
@@ -40,15 +40,18 @@ func NewPeerGater(connGater ConnectionGater, log log.Logger, banEnabled bool) Pe
 func (s *gater) Update(id peer.ID, score float64) {
 	// Check if the peer score is below the threshold
 	// If so, we need to block the peer
+	isAlreadyBlocked := slices.Contains(s.connGater.ListBlockedPeers(), id)
 	if score < PeerScoreThreshold && s.banEnabled {
-		s.log.Warn("peer blocking enabled, blocking peer", "id", id.String(), "score", score)
+		if !isAlreadyBlocked {
```

Comment:
We should be able to skip re-blocking the peer entirely if they're already blocked rather than just skipping the log line.

Also the `Contains(...ListBlockedPeers())` is a pretty expensive check. Inside `connGater` there's actually a nice map keyed by the blocked peer ID, but `ListBlockedPeers` copies all the blocked peers to a list and then we're doing an `O(n)` search of those. If I spam with a lot of unique peer IDs I can potentially burn a lot of CPU here.

We really just want to have a `connGater.IsBlocked(peerId)` method exposed that uses the internal map for fast look up and returns a boolean.

Reply:
Don't have access to the Connection Gater's internal `blockedPeers` mapping, so I set up a sidecar `blockedMap` inside `gater` that is update alongside connection gating.

Let me know what you think of this solution 🙏